### ISSUE TITLE
Bug: [no-extraneous-class] false positive on classes with only index signatures

### ISSUE THREAD

**Issue by undsoft** [typescript-eslint/typescript-eslint#12061]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=MYGwhgzhAECSB2ATApgD2Y6BvAUNaA2gNbICeAXNBAC4BOAlvAOYC6lNDzA3DgL445gAe3g1oqaAF5o8ZAHc4SNBgAUASi5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDJog6xktMvH7JUGSImjQO0SODABfEGqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Indexed {
  [key: string]: string;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-extraneous-class": "error"
  },
};
```

### tsconfig

```jsonc

```

### Expected Result

The class shouldn't be flagged, because it may have non-static members. 
It's an edge case that probably won't happen much in the wild.

### Actual Result

> Unexpected class with only static properties.

**Comment by kirkwaiblinger** [comment]
I guess this isn't explicitly stated on the rule docs as a motivation but this case does unfortunately run into the TS type system pathology where a non-class object can be assigned to the class type: [playground](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=MYGwhgzhAECSB2ATApgD2Y6BvAUNaA2gNbICeAXNBAC4BOAlvAOYC6lNDzA3DgL445gAe3g1oqSghTpMAXmy8uQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDJog6xktMvH7JUGSImjQO0SODABfEGqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false).

Not necessarily a reason not to accept this, just a remark
